### PR TITLE
GCP Batch runnable improvements

### PR DESCRIPTION
**Pull request by AlexITC**
Let's get the implementation closer to lifesciences action, there are still a few details that we couldn't match but those do not seem to be necessary right now.

### CODE REVIEW

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/api/RunnableFlag.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package cromwell.backend.google.pipelines.batch.api
+
+sealed trait RunnableFlag
+
+object RunnableFlag {
+//  case object Unspecified extends RunnableFlag
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/api/RunnableFlag.scala:6]: The commented flags are used right now

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala [hunk 2]
```diff
@@ -45,14 +43,41 @@ object RunnableBuilder {
...
     }
 
+    def withFlags(flags: List[RunnableFlag]): Runnable.Builder = {
+      flags.foldLeft(builder) {
+//        case (acc, RunnableFlag.Unspecified) =>
+        case (acc, RunnableFlag.IgnoreExitStatus) => acc // TODO: How do we handle this?
+        case (acc, RunnableFlag.RunInBackground) => acc.setBackground(true)
+        case (acc, RunnableFlag.AlwaysRun) => acc.setAlwaysRun(true)
+//        case (acc, RunnableFlag.PublishExposedPorts) =>
+//        case (acc, RunnableFlag.DisableImagePrefetch) =>
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala:53]: There are some unused flags that we don't know how to use yet

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala [hunk 3]
```diff
@@ -45,14 +43,41 @@ object RunnableBuilder {
         )
     }
 
+    def withFlags(flags: List[RunnableFlag]): Runnable.Builder = {
+      flags.foldLeft(builder) {
+//        case (acc, RunnableFlag.Unspecified) =>
+        case (acc, RunnableFlag.IgnoreExitStatus) => acc // TODO: How do we handle this?
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala:49]: I haven't found how to set this value

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala [hunk 4]
```diff
@@ -45,14 +43,41 @@ object RunnableBuilder {
...
+    }
+
     def withVolumes(volumes: List[Volume]): Runnable.Builder = {
-      // TODO: Add mount options
-      val formattedVolumes = volumes.map(v => s"${v.getMountPath}:${v.getMountPath}")
+      val formattedVolumes = volumes.map { volume =>
+        val mountPath = volume.getMountPath
+        // TODO: Enable mount options once we know how to test different cases, for now, we keep this disabled
+        //       In theory, the commented code should be enough.
+//        val mountOptions = Option(volume.getMountOptionsList).map(_.asScala.toList).getOrElse(List.empty)
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/runnable/RunnableBuilder.scala:62]: In theory, this should be enough but I haven't tested multiple mount options, better to leave this commented until we verify that it works

**Review by dspeck1 — APPROVED** [review]
looks good

**Pull request merged**